src/uploadIPFS.ts

```diff
@@ -50,26 +76,26 @@ export async function uploadFileIPFS(
   tokens: ApiTokens,
   retries: number = 3
 ): Promise<string> {
-  try {
-    console.log("Uploading file to IPFS...", filePath, 'Retries remaining: ', retries);
-    const fileContents = fs.readFileSync(filePath);
-    const car = await dataToCar(fileContents);
-    return uploadCarFileIPFS(car, tokens);
-  } catch(e) {
-    if (retries > 0) {
-      console.log('Retrying upload', retries);
-      return uploadFileIPFS(filePath, tokens, retries - 1);
-    } else {
-      throw e;
-    }
+  console.log("Uploading file to IPFS....", filePath, 'Retries remaining: ', retries);
+  const fileContents = fs.readFileSync(filePath);
+  const car = await dataToCar(fileContents);
+  const { ipfsHash, error } = await uploadCarFileIPFS(car, tokens);
+  if (!error) return ipfsHash;
+  if (retries > 0) {
+    await sleep(RETRY_DELAY);
```

Comment:
this sleep here should be enough , why we need another sleep and a global rate limited status?

Reply:
Yeah you're right, I'll remove the global rate limited variable